osquery/dispatcher/io_service.cpp

```diff
@@ -0,0 +1,38 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <osquery/logger.h>
+
+#include "osquery/dispatcher/io_service.h"
+
+namespace osquery {
+
+void IOServiceRunner::start() {
+  boost::asio::io_service::work work(IOService::get());
+  for (;;) {
+    try {
+      IOService::get().run();
```

Comment:
This is odd, what happens when this thread ends? The daemon continues to run without the IOService?

Reply:
IOService::get().run(); never return unless IOService::get().stop(); is called